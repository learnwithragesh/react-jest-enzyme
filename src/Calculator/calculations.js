module.exports = {
    add: (a, b) => {
        return parseFloat(a) + parseFloat(b);
    },
    subtract: (a, b) => {
        return parseFloat(a) - parseFloat(b);
    },
    multiply: (a, b) => {
        return parseFloat(a) * parseFloat(b);
    },
    divide: (a, b) => {
        return parseFloat(a) / parseFloat(b);
    },
}