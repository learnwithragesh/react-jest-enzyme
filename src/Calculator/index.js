import React from 'react';
import { useState } from "react";
import { add, subtract, multiply, divide } from './calculations';

const style = {
    margin: {
        margin: 10
    }
}

export default function Calculator() {

    let [number1, setNumber1] = useState('');
    let [number2, setNumber2] = useState('');

    return (
        <div>
            <input type="text" placeholder="Number 1" style={style.margin} 
                value={number1} onChange={(e) => setNumber1(e.target.value)}/>
            <br />
            <input type="text" placeholder="Number 2" style={style.margin}
                value={number2} onChange={(e) => setNumber2(e.target.value)}/>
            <br />
            <input type="button" value="Add" style={style.margin}
                onClick={(e) => alert(add(number1, number2))}/>
            <input type="button" value="Sub" style={style.margin}
                onClick={(e) => alert(subtract(number1, number2))}/>
            <input type="button" value="Mul" style={style.margin}
                onClick={(e) => alert(multiply(number1, number2))}/>
            <input type="button" value="Div" style={style.margin}
                onClick={(e) => alert(divide(number1, number2))}/>
        </div>
    );
}