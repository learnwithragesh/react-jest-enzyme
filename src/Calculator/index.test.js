import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Calculator from './index';

configure({adapter: new Adapter()});

test('Alert Displays on Clicking Add', () => {
    const calculator = shallow(
        <Calculator />,
    );
    expect(calculator.find({ type: 'button' })).toHaveLength(4);
    calculator.find({ type: 'button', value: 'Add'}).simulate('click');
});