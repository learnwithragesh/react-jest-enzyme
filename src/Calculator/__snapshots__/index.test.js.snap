// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Displays on Clicking Add 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Number 1"
    style={
      Object {
        "margin": 10,
      }
    }
    type="text"
    value=""
  />
  <br />
  <input
    onChange={[Function]}
    placeholder="Number 2"
    style={
      Object {
        "margin": 10,
      }
    }
    type="text"
    value=""
  />
  <br />
  <input
    onClick={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    type="button"
    value="Add"
  />
  <input
    onClick={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    type="button"
    value="Sub"
  />
  <input
    onClick={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    type="button"
    value="Mul"
  />
  <input
    onClick={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    type="button"
    value="Div"
  />
</div>
`;
